cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -286,17 +299,29 @@ public String cloneRemote()
     }
     // Clean up the directory if it already exists.
     deletePathIfExists(getRepositoryRoot());
-    RepositoryConfig repositoryConfig = sourceControlConfig
-        .getRepositoryConfig();
-    CloneCommand command =
-        createCommand(Git::cloneRepository).setURI(repositoryConfig.getLink())
-            .setDirectory(getRepositoryRoot().toFile());
+    RepositoryConfig repositoryConfig = sourceControlConfig.getRepositoryConfig();
+    CloneCommand command = createCommand(Git::cloneRepository).setURI(
+        repositoryConfig.getLink()).setDirectory(getRepositoryRoot().toFile());
     String branch = getBranchRefName(repositoryConfig.getDefaultBranch());
     if (branch != null) {
       command.setBranchesToClone(Collections.singleton((branch)))
           .setBranch(branch);
     }
+    long cloneStartTimeMillis = System.currentTimeMillis();
     git = command.call();
+    long cloneTimeMillis = System.currentTimeMillis() - cloneStartTimeMillis;
+    // Record the repository size metric.
+    try {
+      // Prevent division by zero.
+      long repoSize = Math.max(calculateDirectorySize(getRepositoryRoot()), 1);
+      metricsContext.gauge(SourceControlManagement.METRIC_CLONE_LATENCY,
```

Comment:
similar comment here, gauge doesn't seem to fit very well.

Reply:
Switched to using counter metrics.